app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventInitial/EventInitialActivity.java

```diff
@@ -501,6 +501,7 @@ public void setProgramStage(ProgramStage programStage) {
         binding.geometry.setCurrentLocationListener(geometry -> {
             Timber.tag("EVENTINITIAL").d("NEW GEOMETRY");
             this.newGeometry = geometry;
+            preferences.setValue(Preference.EVENT_COORDINATE_CHANGED,true);
```

Comment:
In some cases we access preferences in activity (like this one) in other cases in presenter. I think we should maintaining an architecture order and access data as always from presenter, also having preferences in activity makes to the testing more difficult.

Maybe you could do this in presenter like the other save. In fact, you could have a method name.

fun name(hasCoordinateChange: Boolean){
preferences.setValue(Preference.EVENT_COORDINATE_CHANGED, hasCoordinateChange);
}

thoughts?